packages/manager-components/src/components/public-cloud-project-provider/PublicCloudProjectProvider.tsx

```diff
@@ -0,0 +1,28 @@
+import { createContext, useContext } from 'react';
+
+import { getIsDiscoveryProject } from './util';
+import { PublicCloudProject } from './publicCloudProject';
+
+const PublicCloudProjectContext = createContext(null);
+
+export const PublicCloudProjectProvider = ({
```

Comment:
there is a discussion regarding usage of Context & react-query :
https://github.com/TanStack/query/discussions/3859

i think it *could* make sense for project since it's used basically everywhere in PCI but still i'm not sure if it's worth it when using react query

also this comment is interesting : "One thing to keep in mind is to memoize the children to the Provider to avoid unnecessarily re-rendering the whole tree, and you also cannot do fine grained subscriptions like you can do with e.g. the select option on useQuery. Tradeoffs everywhere "

Reply:
I don't think it's worth it either in our case since they talk about having hundred or thousands of useQuery.
It would make more sense to abstract the useQuery and return utils directly :

```
const query = useQuery(projectQueryData);

return {
   ...query,
   isDiscoveryProject: someFn(data)
};
```

Reply:
I still feel using context can reduce the effort to get the project information except the initial set-up part. Lets say we create the custom hook to get the project details. So whenever you need to get the details, you end up doing the following,

```
import { useParams } from 'react-router-dom';
import { customHook } from 'manager-components';

// inside the component

const { projectId } = useParams();
const { isDiscoveryProject } = customHook(projectId);
```

But with context and provider, the implementation is reduced to half,

```
import { useIsDiscoveryProject }  from 'manager-components';

// inside the component

const isDiscoveryProject = useIsDiscoveryProject();
```

This reduces code duplication.

Reply:
You could call useParams in the customHook from manager-components instead